secretstores/hashicorp/vault/vault.go

```diff
@@ -217,12 +217,12 @@ func (v *vaultSecretStore) BulkGetSecret(req secretstores.BulkGetSecretRequest)
 	// Set X-Vault-Request header
 	httpReq.Header.Set(vaultHTTPRequestHeader, "true")
 	if err != nil {
-		return secretstores.GetSecretResponse{Data: nil}, fmt.Errorf("couldn't generate request: %s", err)
+		return secretstores.BulkGetSecretResponse{Data: nil}, fmt.Errorf("couldn't generate request: %s", err)
```

Comment:
It seems like there's a bug here that this error isn't handled immediately after the call to `NewRequestWithContext`, and could result in a nullref. L216 & L218 both interact with `httpReq` after a possible error.